frontend/src/metabase/dashboard/containers/DashboardApp.jsx

```diff
@@ -82,47 +89,76 @@ const mapDispatchToProps = {
   onChangeLocation: push,
 };
 
-@connect(mapStateToProps, mapDispatchToProps)
-@favicon(({ pageFavicon }) => pageFavicon)
-@title(({ dashboard, documentTitle }) => ({
-  title: documentTitle || dashboard?.name,
-  titleIndex: 1,
-}))
-@titleWithLoadingTime("loadingStartTime")
 // NOTE: should use DashboardControls and DashboardData HoCs here?
-export default class DashboardApp extends Component {
-  state = {
-    addCardOnLoad: null,
-  };
+const DashboardApp = props => {
+  const options = parseHashOptions(window.location.hash);
+  const { loadingComplete, dashboard } = props;
+
+  const [addCardOnLoad] = useState(options.edit);
+  const [editingOnLoad] = useState(options.add && parseInt(options.add));
+
+  const [sendNotification, setSendNotification] = useState(false);
+  const [showToaster, setShowToaster] = useState(false);
+
+  const toastTrigger = useLoadingTimer(!loadingComplete, 5000);
+  const [requestPermission, showNotification] = useWebNotification();
+
+  useEffect(() => {
+    return props.reset;
+  }, [props.reset]);
```

Comment:
You can use [`useOnUnmount` hook](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/hooks/use-on-unmount.ts) to make it a bit more readable